IPython/core/formatters.py

```diff
@@ -422,9 +422,11 @@ def lookup_by_type(self, typ):
                 return self.deferred_printers[typ_key]
         else:
             for cls in pretty._get_mro(typ):
-                if cls in self.type_printers or self._in_deferred_types(cls):
+                try:
                     return self.type_printers[cls]
-        
+                except (KeyError, TypeError):
+                    pass
+
```

Comment:
I'm not entirely sure about this change. In the original code, the if-clause was True even if `cls` is not in `self.type_printers`, but `self._in_deferred_types(cls)` evaluates to True. Doesn't `self.type_printers[cls]` raise `KeyError` in that case? So, that was a bit unclear to me, maybe there's some hidden magic happening and this change isn't correct. I wasn't able to run the tests for this module because I just got a huge bunch of failures on master.